### ISSUE TITLE
Reward Variability

### ISSUE THREAD

**Issue by waybarrios** [om-ai-lab/VLM-R1#113]
Hi,

First of all, thank you so much for this incredible work!

I’ve been trying to replicate the results shown in the figure you published. To do this, I conducted two experiments using 8 H100 GPUs and 8 Mi300X (AMD) GPUs. Both setups used the same datasets, hyperparameters, model checkpoints, and software environment (packages, Python, and PyTorch versions).

## Original reward plot

![Image](https://github.com/user-attachments/assets/77657dfd-374a-4bf9-b53c-c7db474752f9)

## H100
For the H100 GPUs, I obtained the following reward function:

<img width="968" alt="Image" src="https://github.com/user-attachments/assets/f0f16161-a33c-42eb-8619-f9cea49185bd" />

## Mi300x
For the Mi300X GPUs, the results were as follows:

<img width="968" alt="Image" src="https://github.com/user-attachments/assets/e41a2847-67b1-489f-a3ed-1c2c1de1b774" />

## Explanation

In both cases, I noticed a significant variability in the reward function, which wasn’t as apparent in the original graph published in this repository. Additionally, the variability seems slightly more pronounced on the AMD setup.

Since I didn’t modify the code and kept all hyperparameters identical, I’m wondering if there’s something I might be missing or overlooking in the training process. Could you share more details about how you conducted the training? I’d greatly appreciate any insights into the steps you took, as I’d like to ensure I’m replicating the process as accurately as possible.

Thank you very much for your time and assistance!
